floris/simulation/flow_field.py

```diff
@@ -80,36 +90,46 @@ def _discretize_turbine_domain(self):
 
         for i, (coord, turbine) in enumerate(self.turbine_map.items):
 
-            x2, x3 = coord.x2, coord.x3
+            x1, x2, x3 = coord.x1, coord.x2, coord.x3
 
             # Save the indices of the flow field points for this turbine
-            turbine.flow_field_point_indices = i * ngrid * ngrid + np.arange(ngrid * ngrid)
+            turbine.flow_field_point_indices = i * ngrid * ngrid + np.arange(
+                ngrid * ngrid
+            )
 
             pt = turbine.rloc * turbine.rotor_radius
 
             xt = [coord.x1 for coord in self.turbine_map.coords]
             yt = np.linspace(x2 - pt, x2 + pt, ngrid,)
             zt = np.linspace(x3 - pt, x3 + pt, ngrid,)
 
-            for j in range(len(yt)):
-                for k in range(len(zt)):
-                    x_grid[i, j, k] = xt[i]
-                    y_grid[i, j, k] = yt[j]
-                    z_grid[i, j, k] = zt[k]
-
-                    xoffset = x_grid[i, j, k] - coord.x1
-                    yoffset = y_grid[i, j, k] - coord.x2
-                    x_grid[i, j, k] = (
-                        xoffset * cosd(-1 * self.wind_map.turbine_wind_direction[i])
-                        - yoffset * sind(-1 * self.wind_map.turbine_wind_direction[i])
-                        + coord.x1
-                    )
+            x_grid[i] = xt[i]
+            y_grid[i] = yt
+            z_grid[i] = zt
 
-                    y_grid[i, j, k] = (
-                        yoffset * cosd(-1 * self.wind_map.turbine_wind_direction[i])
-                        + xoffset * sind(-1 * self.wind_map.turbine_wind_direction[i])
-                        + coord.x2
-                    )
+            x_grid[i], y_grid[i] = self._update_grid(
```

Comment:
I left this here for review, as I liked Rob's change of making a separate function for `update_grid`, so if you agree @rafmudaf , feel free to remove the commented code.

Reply:
This is fine by me but can we add a docstring to the new function?